src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -837,21 +837,25 @@ private void FreeOverlapped(bool checkForShutdown)
                 if (_socketAddressGCHandle.IsAllocated)
                 {
                     _socketAddressGCHandle.Free();
+                    _pinnedSocketAddress = null;
```

Comment:
We have the same code in .Net Desktop: FreeOverlapped caches the pinned items which will be repinned if different when `PinSocketAddressBuffer()` is called.

If we remove this, is the following return statement still reachable?

```C#
        // Ensures appropriate SocketAddress buffer is pinned.
        private void PinSocketAddressBuffer()
        {
            // Check if already pinned.
            if (_pinnedSocketAddress == _socketAddress)
            {
                return; // < -------
            }

[...]
```